acme_diags/driver/enso_diags_driver.py

```diff
@@ -0,0 +1,168 @@
+
+    for season in seasons:
+        print('Season: {}'.format(season))
+        # Get the name of the data, appended with the years averaged.
+        parameter.test_name_yrs = utils.general.get_name_and_yrs(parameter, test_data, season)
+        parameter.ref_name_yrs = utils.general.get_name_and_yrs(parameter, ref_data, season)
+
+        # Get land/ocean fraction for masking.
+        try:
+            land_frac = test_data.get_climo_variable('LANDFRAC', season)
+            ocean_frac = test_data.get_climo_variable('OCNFRAC', season)
+        except:
+            mask_path = os.path.join(acme_diags.INSTALL_PATH, 'acme_ne30_ocean_land_mask.nc')
+            with cdms2.open(mask_path) as f:
+                land_frac = f('LANDFRAC')
+                ocean_frac = f('OCNFRAC')
+
+        for var in variables:
+            print('Variable: {}'.format(var))
+            parameter.var_id = var
+
+            parameter.viewer_descr[var] = 'Regression coefficient, {} over nino index'.format(var)
+
+            for region in regions:
+                print("Selected region: {}".format(region))
+
+                # This will be the title of the plot.
+                parameter.main_title = ' '.join(
+                    [parameter.viewer_descr[var], season, region])
+
+                # Test
+                test_domain, test_reg_coe = perform_regression(
+                    True, parameter, var, region, land_frac, ocean_frac, test_nino_index, test_sst)
+
+                # Reference
+                ref_domain, ref_reg_coe = perform_regression(
+                    False, parameter, var, region, land_frac, ocean_frac, ref_nino_index, test_sst)
+
+                # Difference
+                # Regrid towards the lower resolution of the two variables for calculating the difference.
+                test_reg_coe_regrid, ref_reg_coe_regrid = utils.general.regrid_to_lower_res(test_reg_coe, ref_reg_coe, parameter.regrid_tool, parameter.regrid_method)
+                diff = test_reg_coe_regrid - ref_reg_coe_regrid
+                metrics_dict = create_metrics(ref_reg_coe, test_reg_coe, ref_reg_coe_regrid, test_reg_coe_regrid, diff)
+
+                # Plot
+                parameter.var_region = region
+                # Call acme_diags.__init__/plot, which uses
+                # acme_diags.__init__/_get_plot_fcn to call
+                # acme_diags.plot.{backend}.{set_name}_plot
+                # {parameter.backend} defaults to `mpl`
+                # (in acme_diags.parameter.core_parameter.CoreParameter.__init__),
+                # which will use the `cartopy` backend.
+                # `set_name` is set to `parameter.curent_set` below.
+                # `parameter.current_set` is set to the current set from
+                # `parameters.sets` in acme_diags.acme_diags_driver.run_diag
+                # e.g. acme_diags.plot.cartopy.enso_diags_plot is called
+                plot(parameter.current_set, ref_reg_coe_regrid,
```

Comment:
This `plot` function was used when for each plot set, we supported both `vcs` and `matplotlib`. I think sadly this isn't the case anymore. You might want to just import `enso_diags_plot.py` and use the plot function directly. This makes it more clear to a reader.